### PR TITLE
[velero] feat: add annotations for velero deployment and restic daemonset

### PR DESCRIPTION
**Pull request by cyril-corbon**
#### Special notes for your reviewer:
This PR adds:

    Ability to set annotation and label on Velero deployment
    Ability to set annotation and label on Restic daemonset
    Ability to set annotation and label on velero service

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the values.yaml or README.md
- [x] Title of the PR starts with chart name (e.g. `[velero]`)

### CODE REVIEW

**Comment by jenting** [comment]
Related to #194

File: charts/velero/templates/deployment.yaml [hunk 1]
```diff
@@ -4,6 +4,10 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: {{ include "velero.fullname" . }}
+  annotations:
+    {{- with .Values.deploymentAnnotations }}
+      {{- toYaml . | nindent 6 }}
+    {{- end }}
```
Comment (jenting) [inline charts/velero/templates/deployment.yaml:10]:
```yaml
{{- if .Values.deploymentAnnotations }}
  annotations:
  {{- with .Values.deploymentAnnotations }}
    {{- toYaml . | nindent 4 }}
  {{- end }}
{{- end }}
```

File: charts/velero/templates/restic-daemonset.yaml [hunk 2]
```diff
@@ -4,6 +4,10 @@ apiVersion: apps/v1
 kind: DaemonSet
 metadata:
   name: restic
+  annotations:
+    {{- with .Values.restic.daemonsetAnnotations }}
+      {{- toYaml . | nindent 6 }}
+    {{- end }}
```
Comment (jenting) [inline charts/velero/templates/restic-daemonset.yaml:10]:
```yaml
{{- if .Values.restic.daemonsetAnnotations }}
  annotations:
  {{- with .Values.restic.daemonsetAnnotations }}
    {{- toYaml . | nindent 4 }}
  {{- end }}
{{- end }}
```

**Review by jenting — CHANGES_REQUESTED** [review]
Suggestion changes.

File: charts/velero/templates/deployment.yaml (hunk 1 above)
Comment (cyril-corbon) [reply charts/velero/templates/deployment.yaml:11]: Thanks for the suggestion, I just applied it.

Comment (jenting) [reply charts/velero/templates/deployment.yaml:11]: I just realized that we could make it simpler.
```yaml
  {{- with .Values.deploymentAnnotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```

File: charts/velero/templates/restic-daemonset.yaml (hunk 2 above)
Comment (jenting) [reply charts/velero/templates/restic-daemonset.yaml:11]: I just realized that we could make it simpler.
```yaml
  {{- with .Values.restic.daemonsetAnnotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```

**Review by jenting — CHANGES_REQUESTED** [review]

**Comment by cyril-corbon** [comment]
I Just applied the review and add labels / annotations to Deployments /  Daemonset / Service

**Comment by cyril-corbon** [comment]
Is there any blocking point preventing you from merging this PR? What can I do for it to be merged ?

**Comment by jenting** [comment]
> Is there any blocking point preventing you from merging this PR? What can I do for it to be merged ?

Sorry for the delay review, I'll check it today. Thank u.

**Comment by jenting** [comment]
Would you mind to bump the chart version? Thank u.

**Comment by cyril-corbon** [comment]
@jenting I just done it :smile:

**Review by jenting — APPROVED** [review]
LGTM, thank again.

**Comment by jenting** [comment]
@cpanato @carlisia PTAL 🙇‍♂️

**Review by cpanato — APPROVED** [review]
/lgtm

**Pull request merged**